helm-service/controller/deployment_handler.go

```diff
@@ -212,27 +202,39 @@ func (h *DeploymentHandler) getStartedEventData(inEventData keptnv2.EventData) k
 	return keptnv2.DeploymentStartedEventData{EventData: inEventData}
 }
 
-func (h *DeploymentHandler) getFinishedEventDataForSuccess(inEventData keptnv2.EventData, gitCommit string,
+func (h *DeploymentHandler) getFinishedEventDataForSuccess(inEventData keptnv2.DeploymentTriggeredEventData, gitCommit string,
 	deploymentName string, deploymentStrategy keptnevents.DeploymentStrategy) (*keptnv2.DeploymentFinishedEventData, error) {
 
 	inEventData.Status = keptnv2.StatusSucceeded
 	inEventData.Result = keptnv2.ResultPass
 	inEventData.Message = "Successfully deployed"
 
-	localURIs, publicURIs, err := h.getDeploymentURIs(inEventData)
-	if err != nil {
-		return nil, fmt.Errorf("could not determine deployment URIs: %s", err.Error())
+	var localURIs, publicURIs []string
+	if len(inEventData.Deployment.DeploymentURIsLocal) > 0 || len(inEventData.Deployment.DeploymentURIsPublic) > 0 {
+		h.getKeptnHandler().Logger.Info("Using deployment URIs from deployment.triggered event")
+		localURIs = inEventData.Deployment.DeploymentURIsLocal
+		publicURIs = inEventData.Deployment.DeploymentURIsPublic
+	} else if deploymentStrategy == keptnevents.Direct || deploymentStrategy == keptnevents.Duplicate {
+		h.getKeptnHandler().Logger.Info("Inferring deployment URIs")
+		var err error
+		localURIs, publicURIs, err = h.getDeploymentURIs(inEventData.EventData)
+		if err != nil {
+			return nil, fmt.Errorf("Could not determine deployment URIs: %s", err.Error())
+		}
+	} else {
+		h.getKeptnHandler().Logger.Info("No deployment URIs defined in deployment.finished event")
```

Comment:
Won't this lead to problems if the incoming event contains deployment URIs but deploymentstrategy is direct or duplicate?

Reply:
I do not think so because the URLs are still inferred for direct and duplicate in case they are not passed in the incoming triggered event.

Reply:
> in case they are not passed in the incoming triggered event.

That's exactly what I asked. So this code needs to be changed.

Reply:
Just verified, leads to the problem I expected: production stage reports dev url
```
    {
        "data": {
            "deployment": {
                "deploymentNames": [
                    "canary"
                ],
                "deploymentURIsLocal": [
                    "http://carts.sockshop-new-dev:80"
                ],
                "deploymentURIsPublic": [
                    "http://carts.sockshop-new-dev.34.69.52.253.nip.io:80"
                ],
                "deploymentstrategy": "duplicate",
                "gitCommit": "1fcfa771559ca8dc2e3378b0cf9a074ec4119d88"
            },
            "message": "Successfully deployed",
            "project": "sockshop-new",
            "result": "pass",
            "service": "carts",
            "stage": "production",
            "status": "succeeded"
        },
```